profiling/src/config.rs

```diff
@@ -200,39 +202,49 @@ pub(crate) unsafe fn profiling_enabled() -> bool {
     get_bool(ProfilingEnabled, true)
 }
 
+/// # Safety
+/// This function must only be called after config has been initialized in
+/// rinit, and before it is uninitialized in mshutdown.
+pub(crate) unsafe fn profiling_experimental_features_enabled() -> bool {
+    profiling_enabled() && get_bool(ProfilingExperimentalFeaturesEnabled, false)
+}
+
 /// # Safety
 /// This function must only be called after config has been initialized in
 /// rinit, and before it is uninitialized in mshutdown.
 pub(crate) unsafe fn profiling_endpoint_collection_enabled() -> bool {
-    get_bool(ProfilingEndpointCollectionEnabled, true)
+    profiling_enabled() && get_bool(ProfilingEndpointCollectionEnabled, true)
 }
 
 /// # Safety
 /// This function must only be called after config has been initialized in
 /// rinit, and before it is uninitialized in mshutdown.
 pub(crate) unsafe fn profiling_experimental_cpu_time_enabled() -> bool {
-    get_bool(ProfilingExperimentalCpuTimeEnabled, true)
+    profiling_experimental_features_enabled()
+        || profiling_enabled() && get_bool(ProfilingExperimentalCpuTimeEnabled, true)
```

Comment:
Shouldn't it be `profiling_enabled() && (profiling_experimental_features_enabled() || get_bool(ProfilingExperimentalCpuTimeEnabled, true))`?

Wait, `profiling_experimental_features_enabled()` also does `profiling_enabled()` so we're checking that twice, but this makes them equivalent. It still seems easier in my brain to do it the way I listed out, though, because `profiling_enabled` is the highest priority, so it makes sense to put that first... right?